src/textual/css/_styles_builder.py

```diff
@@ -252,6 +252,23 @@ def scalar_error():
         else:
             scalar_error()
 
+    def _distribute_importance(self, prefix: str, suffixes: tuple[str, ...]) -> None:
+        """Distribute importance amongst all aspects of the given style.
+
+        Args:
+            prefix: The prefix of the style.
+            siffixes: The suffixes to distribute amongst.
+
+        A number of styles can be set with the 'prefix' of the style,
+        providing the values as a series of parameters; or they can be set
+        with specific suffixes. Think `border` vs `border-left`, etc. This
+        method is used to ensure that if the former is set, `!important` is
+        distributed amongst all the suffixes.
+        """
+        if prefix in self.styles.important:
+            self.styles.important.remove(prefix)
+            self.styles.important.update(f"{prefix}_{suffix}" for suffix in suffixes)
```

Comment:
```suggestion
        self.styles.important.discard(prefix)
        self.styles.important.update(f"{prefix}_{suffix}" for suffix in suffixes)
```

The discard will save a lookup. I think `prefix` is always going to be in there anyway?

Reply:
The check is the check that the style is `!important`, otherwise we don't want all the suffixes marked as important.